whois-api/src/test/java/net/ripe/db/whois/api/rest/WhoisRestServiceDoSTestIntegration.java

```diff
@@ -0,0 +1,249 @@
+package net.ripe.db.whois.api.rest;
+
+import jakarta.ws.rs.client.Entity;
+import jakarta.ws.rs.client.Invocation;
+
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.Response;
+import net.ripe.db.whois.api.AbstractIntegrationTest;
+import net.ripe.db.whois.api.RestTest;
+import net.ripe.db.whois.api.rest.domain.WhoisResources;
+import net.ripe.db.whois.api.rest.mapper.FormattedClientAttributeMapper;
+import net.ripe.db.whois.api.rest.mapper.WhoisObjectMapper;
+import net.ripe.db.whois.common.rpsl.RpslObject;
+import org.eclipse.jetty.http.HttpStatus;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.IntStream;
+
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+
+@Tag("IntegrationTest")
+public class WhoisRestServiceDoSTestIntegration extends AbstractIntegrationTest {
+
+    @Autowired
+    private WhoisObjectMapper whoisObjectMapper;
+
+    private static final RpslObject OWNER_MNT = RpslObject.parse("" +
+            "mntner:      OWNER-MNT\n" +
+            "descr:       Owner Maintainer\n" +
+            "admin-c:     TP1-TEST\n" +
+            "upd-to:      noreply@ripe.net\n" +
+            "auth:        MD5-PW $1$d9fKeTr2$Si7YudNf4rUGmR71n/cqk/ #test\n" +
+            "auth:        SSO person@net.net\n" +
+            "mnt-by:      OWNER-MNT\n" +
+            "source:      TEST");
+
+    private static final Integer PENULTIMATE_OK_UPDATE_POSITION = 8;
+
+    private static final Integer PENULTIMATE_OK_LOOKUP_POSITION = 48;
+
+    private static final Integer MAXIMUM_UPDATE_REQUESTS_ALLOWED_PER_SECOND = 10;
+
+    private static final Integer MAXIMUM_LOOKUP_REQUESTS_ALLOWED_PER_SECOND = 50;
+
+    private static final Integer SECONDS_NEEDED_TO_FREE_IP = 1;
+
+    @BeforeAll
+    public static void beforeClass() {
```

Comment:
clear the list or set the dos filter false in AfterEach or AfterAll?

Reply:
Done